app/data_test.go

```diff
@@ -0,0 +1,616 @@
+		test.That(t, response, test.ShouldResemble, expectedRawData)
+	})
+
+	t.Run("BinaryDataByFilter", func(t *testing.T) {
+		includeBinary := true
+		countOnly := true
+		includeInternalData := true
+		dataRequest := DataRequest{
+			Filter:    filter,
+			Limit:     5,
+			Last:      "last",
+			SortOrder: Unspecified,
+		}
+		expectedCount := uint64(5)
+		expectedLast := "last"
+		expectedBinaryData := BinaryData{
+			Binary:   binaryDataByte,
+			Metadata: binaryMetadata,
+		}
+		expectedBinaryDataPb := binaryDataToProto(expectedBinaryData)
+
+		grpcClient.BinaryDataByFilterFunc = func(ctx context.Context, in *datapb.BinaryDataByFilterRequest,
+			opts ...grpc.CallOption,
+		) (*datapb.BinaryDataByFilterResponse, error) {
+			expectedDataReq := dataRequestToProto(dataRequest)
+			test.That(t, in.DataRequest, test.ShouldResemble, expectedDataReq)
+			test.That(t, in.IncludeBinary, test.ShouldBeTrue)
+			test.That(t, in.CountOnly, test.ShouldBeTrue)
+			test.That(t, in.IncludeInternalData, test.ShouldBeTrue)
+			return &datapb.BinaryDataByFilterResponse{
+				Data:  []*datapb.BinaryData{expectedBinaryDataPb},
+				Count: expectedCount,
+				Last:  expectedLast,
+			}, nil
+		}
+		respBinaryData, respCount, respLast, _ := client.BinaryDataByFilter(
+			context.Background(), filter, expectedCount, dataRequest.SortOrder,
+			expectedLast, includeBinary, countOnly, includeInternalData)
+		test.That(t, respBinaryData[0], test.ShouldResemble, expectedBinaryData)
+		test.That(t, respCount, test.ShouldEqual, expectedCount)
+		test.That(t, respLast, test.ShouldEqual, expectedLast)
+	})
+	t.Run("BinaryDataByIDs", func(t *testing.T) {
+		expectedBinaryData := BinaryData{
+			Binary:   binaryDataByte,
+			Metadata: binaryMetadata,
+		}
+		grpcClient.BinaryDataByIDsFunc = func(ctx context.Context, in *datapb.BinaryDataByIDsRequest,
+			opts ...grpc.CallOption,
+		) (*datapb.BinaryDataByIDsResponse, error) {
+			test.That(t, in.IncludeBinary, test.ShouldBeTrue)
+			test.That(t, in.BinaryIds, test.ShouldResemble, binaryIDsToProto(binaryIDs))
+			expectedBinaryDataList := []*datapb.BinaryData{binaryDataToProto(expectedBinaryData)}
+
+			return &datapb.BinaryDataByIDsResponse{Data: expectedBinaryDataList, Count: uint64(len(expectedBinaryDataList))}, nil
+		}
+		respBinaryData, _ := client.BinaryDataByIDs(context.Background(), binaryIDs)
+		test.That(t, respBinaryData[0], test.ShouldResemble, expectedBinaryData)
+	})
+
+	t.Run("DeleteTabularData", func(t *testing.T) {
+		deleteOlderThanDays := uint32(1)
+		expectedOrgID := organizationID
+		expectedDeleteOlderThanDays := deleteOlderThanDays
+		expectedDeletedCount := uint64(5)
+
+		grpcClient.DeleteTabularDataFunc = func(ctx context.Context, in *datapb.DeleteTabularDataRequest,
+			opts ...grpc.CallOption,
+		) (*datapb.DeleteTabularDataResponse, error) {
+			test.That(t, in.OrganizationId, test.ShouldEqual, expectedOrgID)
```

Comment:
Why not just test if it's equal to `organizationID` directly? This is a question that applies to the other expected variables as well.

Reply:
I was doing it this way because in this test we are expecting to get back the protobuf version of that variable and sometimes the protobuf version is the same thing (like when it is a string or int) but other times it is different and I would pass the original variable through a "ToProto" function. So, because I was sometimes passing it thru a function and othertimes not, I was just trying to stay consistent. I also did this to help me understand exactly what I was testing but I can def change it if its repetitive or makes it more unclear.

Reply:
It is usually bad practice to create a string that's just a repeat and is already pretty well defined. It adds repetition and unnecessary confusion. I do understand what you mean though about consistency and such, but strings and ints should already be clear.